tests/autoyast/installation.pm

```diff
@@ -392,7 +395,7 @@ sub run {
             return;             # Popup itself is processed in opensuse_welcome module
         }
         elsif (match_has_tag('encrypted-disk-password-prompt')) {
-            return;
+            return;             # may be necessary if "bios-boot" matched earlier instead of encrypted-disk-password-prompt
```

Comment:
I don't get this comment, the first `until` (first stage) will finish with any of those two needles and the second `until` (second stage) will return if finds the encrypted one.
So in my understanding you finish logic for first stage when grub appears in those two ways and then you early exit second stage when encrypted, but bios-boot is not included in the second stage needle matching process.
Would be enough to just mention in the comment about the situation with the encrypted?

Reply:
Removed, indeed this needle will match in either case, not just when bios-boot matched.